frontend/components/RawHtml/RawHtml.tsx

```diff
@@ -1,8 +1,96 @@
-import React from "react";
+import React, { Fragment, useEffect, useState, useRef } from "react";
+import { XMarkIcon } from "@heroicons/react/24/outline";
 import s from "./RawHtml.module.css";
+import HolonButton from "../VersionOne/Buttons/HolonButton";
+import { Dialog, Transition } from "@headlessui/react";
 
-const RawHtml = ({ html }: { html: string }) => (
-  <div className={s.Container} dangerouslySetInnerHTML={{ __html: html }} />
-);
+export default function RawHtml({ html }: { html: string }) {
+  const [isOpen, setIsOpen] = useState(true);
+  const [modalText, setModalText] = useState("");
+  const [modalLink, setModalLink] = useState("");
 
-export default RawHtml;
+  const RawHtmlItem = useRef(null);
+
+  function closeModal() {
+    setIsOpen(false);
+    setModalText("");
+    setModalLink("");
+  }
+
+  function openModal(text, linkUrl) {
+    setModalLink(linkUrl);
+    setModalText(text);
+    setIsOpen(true);
+  }
+
+  useEffect(() => {
+    const staticTermLinks = RawHtmlItem.current.querySelectorAll(`a[data-introduction-text]`);
+
+    for (let i = 0; i < staticTermLinks.length; i++) {
+      const link = staticTermLinks[i] as HTMLElement | null;
+      if (link) {
+        link.addEventListener("click", function (e) {
+          e.preventDefault();
+
+          const linkText = link.getAttribute("data-introduction-text");
+          const linkUrl = link.getAttribute("data-page-link");
+          openModal(linkText, linkUrl);
+        });
+      }
+    }
+  }, []);
+
+  return (
+    <Fragment>
+      <div ref={RawHtmlItem} className={s.Container} dangerouslySetInnerHTML={{ __html: html }} />
+      {modalText && (
+        <Transition appear show={isOpen} as={Fragment}>
+          <Dialog as="div" className="relative z-10" onClose={closeModal}>
```

Comment:
This z-index is too low. The header has a value of 50, and if the viewport is small this causes the modal to be overlapped by the header:

<img width="518" alt="Screenshot 2022-11-17 at 09 39 45" src="https://user-images.githubusercontent.com/4383/202412620-4f2f501c-98a9-467c-95c4-0cada1c86833.png">

```suggestion
          <Dialog as="div" className="relative z-50" onClose={closeModal}>
```